src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,104 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd = () => {} }) => {
+  const [count, setCount] = useState(0);
+
+  function createNewMovie() {
+    return {
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    };
+  }
+
+  const [newMovie, setNewMovie] = useState(createNewMovie());
+
+  const resetFormFields = () => {
+    setNewMovie(createNewMovie());
+  };
+
+  const isDisabled = () => {
+    return !(
+      newMovie.title
+      && newMovie.imgUrl
+      && newMovie.imdbUrl
+      && newMovie.imdbId
+    );
+  };
```

Comment:
```suggestion
  const isDisabled = !(
      newMovie.title
      && newMovie.imgUrl
      && newMovie.imdbUrl
      && newMovie.imdbId
    );
```
You don't need to do this check with a function, just use it as a variable